include/dlaf/tile.h

```diff
@@ -0,0 +1,26 @@
+//
```

Comment:
Rename to `include/dlaf/blas.h`?

Reply:
Do you see any plus in keeping it separate? Otherwise, I would go for keeping them all together.

Reply:
No reason other than that the blas and cublas functionality have so far stayed mostly in separate files. This is the current `cublas_tile.h`: https://github.com/eth-cscs/DLA-Future/blob/e70c6518459bf1a0159d94660156e23380a07bf8/include/dlaf/cublas_tile.h. I could move the cublas declarations to `blas_tile.h` as well, and move the cublas definitions to `blas_tile.tpp`. In that case, should things like `util_blas.h` and `util_cublas.h` also be combined?

Reply:
What about merging `include/dlaf/tile.h``include/dlaf/blas_tile.h` and `include/dlaf/cublas_tile.h` in a single file `include/dlaf/blas/tile.h`?
I would anyway keep `util_blas.h` and `util_cublas.h` separate and move them in `include/dlaf/blas` as well.

Reply:
> What about merging `include/dlaf/tile.h`, `include/dlaf/blas_tile.h`, `include/dlaf/blas_tile.tpp` and `include/dlaf/cublas_tile.h` in a single file `include/dlaf/blas/tile.h`?

Ok with me. @albestro?

> I would anyway keep `util_blas.h` and `util_cublas.h` separate and move them in `include/dlaf/blas` as well.

Separately into `dlaf/blas/util.h` and `dlaf/cublas/util.h`?

I suppose there are related renamings that could be done for `lapack_tile.h`, `util_math.h`, `util_matrix.h`, and `util_tile.h`, but I won't move those in this PR.